core/server/common/src/main/java/alluxio/master/PrimarySelectorClient.java

```diff
@@ -35,59 +38,50 @@
  * Masters use this client to elect a leader.
  */
 @NotThreadSafe
-public final class LeaderSelectorClient implements Closeable, LeaderSelectorListener {
-  private static final Logger LOG = LoggerFactory.getLogger(LeaderSelectorClient.class);
+public final class PrimarySelectorClient
+    implements Closeable, LeaderSelectorListener, PrimarySelector {
+  private static final Logger LOG = LoggerFactory.getLogger(PrimarySelectorClient.class);
 
   /** The election path in Zookeeper. */
   private final String mElectionPath;
   /** The path of the leader in Zookeeper. */
   private final String mLeaderFolder;
-  /** The LeaderSelector used to elect. */
+  /** The PrimarySelector used to elect. */
   private final LeaderSelector mLeaderSelector;
   /** The name of this master in Zookeeper. */
-  private final String mName;
+  private String mName;
   /** The address to Zookeeper. */
   private final String mZookeeperAddress;
 
   /** Whether this master is the leader master now. */
   private AtomicBoolean mIsLeader = new AtomicBoolean(false);
-  /** The thread of the current master. */
-  private volatile Thread mCurrentMasterThread = null;
```

Comment:
We were using this to kick the master thread when leadership changed so that we would gain or lose primacy. This has been changed to polling the `mIsLeader` field every 100ms from `FaultTolerantAlluxioMasterProcess`. This way implementations of `PrimarySelectorClient` only need to expose a boolean `isPrimary` method, instead of needing to do their own polling and callbacks to the master process thread.